### ISSUE TITLE
Mistimed Subtitles

### ISSUE THREAD

**Issue by jonathanfox5** [octimot/StoryToolkitAI#125]
**Describe the bug**
After working correctly for several minutes, generated subtitles appear up to 10 seconds early, resulting in significantly misaligned subtitles for approx 30 seconds. After these 30 seconds, the subtitles suddenly fix themselves and become aligned again.

- This seems to happen when there is a period of silence.
- The incorrect timing appears in both the SRT and JSON files.
- This is using standalone mode with no connection to resolve
- Transcription language is set to Italian

**To Reproduce**
1. Example video: https://www.youtube.com/watch?v=oX30MwXrdwM
2. Download using highest quality settings
3. File | Transcribe Audio Files | Select downloaded video
4. Transcribe using default settings (e.g. medium model). Set language to Italian
5. Watch the video with the new subtitles. (I use mpv but this is not a player issue - the issue can be seen by viewing the SRT and JSON files in a text editor)

**Expected behavior**
Subtitles should line up _approximately_ with when they are spoken rather than being 10 seconds out.

In this example, the line "Ciao Lorenzo" appears on screen at 1:55 but the line is not spoken until 2:05. Subtitles are significantly misaligned (and therefore cannot be followed) until 2:25 when they suddenly fix themselves. 

**Log file**
See bottom of this post

**System (please complete the following information):**
 - OS: M2 Macbook, Ventura 13.4.1
 - Python versions installed on machine: Running with 3.10 as suggest
 - StoryToolkitAI Version 0.19.4, using git version rather than standalone.

**Additional context**
Frame rate of the example video is exactly 30 fps. This behaviour occurs in videos from other sources too.

SRT
[La Mappa Misteriosa #1 (ITALIAN).mp4.srt.txt](https://github.com/octimot/StoryToolkitAI/files/12251168/La.Mappa.Misteriosa.1.ITALIAN.mp4.srt.txt)

JSON
[La Mappa Misteriosa #1 (ITALIAN).mp4.transcription.json.txt](https://github.com/octimot/StoryToolkitAI/files/12251201/La.Mappa.Misteriosa.1.ITALIAN.mp4.transcription.json.txt)

app.log
[app.log](https://github.com/octimot/StoryToolkitAI/files/12251312/app.log)

**Comment by jonathanfox5** [comment]
I've been changing settings to see if I can fix it.

Turning on pre-detect speech and increased precision mostly fixes the issue. Interestingly, the line that broke it previously ("Ciao Lorenzo") no longer appears in the captions at all. However, everything that comes after it is now aligned.

I've included an extract below of the new SRT file. The line should appear just before 27. 

```
26
00:01:49,040 --> 00:01:49,680
Vieni anche tu?

27
00:02:05,144 --> 00:02:06,364
Ciao Renato.
```

**Comment by octimot** [comment]
Hey there!

Shortly, this is just AI going a bit weird, so it's not something that we can really control from the tool itself.

Have you tried the large model? How is that performing?

Cheers!

**Comment by jonathanfox5** [comment]
Thanks, the large V2 model deals with it perfectly when every other setting is set to default. The large model cuts out different lines when "detect speech" is selected so I've kept it off.

I've also discovered that changing the initial prompt to match the language selected prevents some other weirdness happening in other videos (incomplete transcriptions, random sentences translated to English, etc.) Maybe changing the prompt based upon the selected language is something worth considering as a low priority feature?

I've therefore settled on the following settings:
<img width="502" alt="image" src="https://github.com/octimot/StoryToolkitAI/assets/32023524/7adef3f2-b111-4304-a384-c4dc46cfed87">

Thanks again for the work on this. This tool has made things a million times easier for me :)
